cfme-performance/utils/providers.py

```diff
@@ -286,6 +287,20 @@ def add_provider(provider):
 
     logger.debug('Added Provider: {}, Response: {}'.format(provider['name'], response))
 
+    # Workaround for Bug: https://bugzilla.redhat.com/show_bug.cgi?id=1351253
+    if provider['type'] == "ManageIQ::Providers::Microsoft::InfraManager":
+        time.sleep(10)
+        logger.debug('Running rails command for Microsoft InfraManager Workaround')
+        ssh_client = SSHClient()
+        command = ('m = ExtManagementSystem.find_by_name "{}"; attributes = {{:security_protocol =>'
+        ' "{}"}}; m.update_authentication(:default => {{:userid => "{}", :password => "{}"}}); '
+        'm.update_attributes(attributes); m.save; m.authentication_check;'
+        .format(provider['name'], provider['security_protocol'],
+            provider['credentials']['username'].replace('\\', '\\\\'),
+            provider['credentials']['password']))
+        ssh_client.run_command("cd /var/www/miq/vmdb; echo '" + command + "'"
+            "| bundle exec bin/rails c 2> /dev/null")
```

Comment:
Matt I modified it for readability and found I could add the provider without the sleep statement through I am still concerned of a race condition in "slower" environments, but I'm not entire sure that is an issue yet.
```python
    # Workaround for Bug: https://bugzilla.redhat.com/show_bug.cgi?id=1351253
    if provider['type'] == "ManageIQ::Providers::Microsoft::InfraManager":
        logger.debug('Running rails command for Microsoft InfraManager Workaround')
        ssh_client = SSHClient()
        command = (
            'm = ExtManagementSystem.find_by_name \'{}\';'
            'attributes = {{:security_protocol => \'{}\'}};'
            'm.update_authentication(:default => {{:userid => \'{}\', :password => \'{}\'}});'
            'm.update_attributes(attributes);'
            'm.save;'
            'm.authentication_check;'.format(provider['name'], provider['security_protocol'],
                provider['credentials']['username'].replace('\\', '\\\\'),
                provider['credentials']['password']))
        ssh_client.run_rails_console(command, timeout=None, log_less=True)
```